src/charm.py

```diff
@@ -70,6 +78,27 @@ def __init__(self, *args):
             extra_sans_dns=[self._hostname],
         )
 
+        # Request an ingress url
+        self._ingress = IngressPerAppRequirer(
+            self, port=9091, scheme=lambda: "https" if self._cert_handler.enabled else "http"
+        )
+
+        # add itself as an entry to the catalogue
+        self._catalogue = CatalogueConsumer(
+            self,
+            item=CatalogueItem(
+                "Prometheus Pushgateway",
+                self._ingress.url or self._hostname + ":9091",
+                icon="chart-timeline-variant-shimmer",
+                description="Prometheus Pushgateway UI.",
+            ),
+            refresh_event=[
+                self._ingress.on.ready,
+                self._ingress.on.revoked,
+                self.on.pushgateway_pebble_ready,
```

Comment:
we probably need to do this on config_changed as well to cater for charm upgrades?

Reply:
charm upgrade will trigger pebble-ready, so that path should be covered.